skfp/descriptors/constitutional.py

```diff
@@ -0,0 +1,252 @@
+    >>> element_atom_count(mol, "H")
+    8
+    >>> element_atom_count(mol, 1)
+    8
+    """
+    if atom_id == 1 or atom_id == "H":
+        return sum(atom.GetTotalNumHs() for atom in mol.GetAtoms())
+    else:
+        return sum(
+            1
+            for atom in mol.GetAtoms()
+            if atom.GetAtomicNum() == atom_id or atom.GetSymbol() == atom_id
+        )
+
+
+@validate_molecule
+def molecular_weight(mol: Mol) -> float:
+    """
+    Molecular Weight.
+
+    Calculates the molecular weight of the molecule [1]_.
+
+    This is average molecular weight in terms of isotopes [2]_.
+
+    Parameters
+    ----------
+    mol : RDKit Mol object
+        The molecule for which the molecular weight is to be calculated.
+
+    References
+    ----------
+    .. [1] `Hiromasa, Kaneko.
+        "Molecular Descriptors, Structure Generation,
+        and Inverse QSAR/QSPR Based on SELFIES"
+        ACS Omega 8.24 (2023): 21781–21786.
+        <https://doi.org/10.1021/acsomega.3c01332>`_
+
+    .. [2] `
+        <https://chemistry.stackexchange.com/questions/150993/discrepancy-when-calulating-mol-weights-with-chemsketch-and-python-rdkit/150999#150999>`_
+
+    Examples
+    --------
+    >>> from rdkit.Chem import MolFromSmiles
+    >>> from skfp.descriptors.constitutional import molecular_weight
+    >>> mol = MolFromSmiles("C1=CC=CC=C1")  # Benzene
+    >>> molecular_weight(mol)
+    78.114
+    """
+    return MolWt(mol)
+
+
+@validate_molecule
+def number_of_rings(mol: Mol) -> int:
+    """
+    Number of rings.
+
+    Calculates the total number of rings in the molecule [1]_.
+
+    Parameters
+    ----------
+    mol : RDKit Mol object
+        The molecule for which the number of rings is to be calculated.
+
+    References
+    ----------
+    .. [1] `Alan H. Lipkus.
+        "Exploring Chemical Rings in a Simple Topological-Descriptor Space"
+        Journal of Chemical Information and Computer Sciences 41.2 (2001): 430–438.
+        <https://doi.org/10.1021/ci000144x>`_
+
+    Examples
+    --------
+    >>> from rdkit.Chem import MolFromSmiles
+    >>> from skfp.descriptors.constitutional import number_of_rings
+    >>> mol = MolFromSmiles("C1=CC=CC=C1")  # Benzene
+    >>> number_of_rings(mol)
+    1
+    """
+    return CalcNumRings(mol)
+
+
+@validate_molecule
+def number_of_rotatable_bonds(mol: Mol) -> int:
+    """
+    Number of rotatable bonds.
+
+    Calculates the total number of rotatable bonds in the molecule.
+
+    Parameters
+    ----------
+    mol : RDKit Mol object
+        The molecule for which the number of rotatable bonds is to be calculated.
+
+    Examples
+    --------
+    >>> from rdkit.Chem import MolFromSmiles
+    >>> from skfp.descriptors.constitutional import number_of_rotatable_bonds
+    >>> mol = MolFromSmiles("C=CC=C")  # Butadiene
+    >>> number_of_rotatable_bonds(mol)
+    1
+    """
+    return CalcNumRotatableBonds(mol)
+
+
+@validate_molecule
+def total_atom_count(mol: Mol) -> int:
```

Comment:
A function for just counting heavy atoms (non-hydrogens) would definitely be useful.